functions/zoom-downloader.py

```diff
@@ -284,59 +280,39 @@ def stream_file_to_s3(file, uuid, track_sequence):
     stream.close()
 
 
-def retrieve_url_from_play_page(play_url):
-
-    r = requests.get(play_url)
-    r.raise_for_status()
-
-    only_source_tags = SoupStrainer("source", type="video/mp4")
-
-    source = BeautifulSoup(r.content, "html.parser", parse_only=only_source_tags)
-    logger.debug(str(source))
+def get_stream(download_url):
 
-    source_object = source.find("source")
-
-    if source_object is None:
-        password_form = BeautifulSoup(r.content, "html.parser",
-                                      parse_only=SoupStrainer(id="password_form"))
-        if len(password_form) == 0:
-            raise PermanentDownloadError("No source element found on page: {}".format(play_url))
-        else:
-            raise PermanentDownloadError("Password protected play url: {}".format(play_url))
-
-    link = source_object['src']
-    logger.info("Got download url {}".format(link))
-
-    return link
+    admin_token = get_admin_token()
 
+    # use zak token to get download stream
+    logger.info("requesting {}".format(download_url))
 
-def get_connection_using_play_url(play_url):
-    file_url = retrieve_url_from_play_page(play_url)
+    r = requests.get("{}?zak={}".format(download_url, admin_token), allow_redirects=False)
+    if 'Location' in r.headers:
+        location = r.headers['Location']
+        zoom_name = location.split('?')[0].split('/')[-1]
+    else:
```

Comment:
so the zoom behavior is not consistent here?

Reply:
File name is in 'Content Disposition' for .vtt (transcript files) but 'Location' for .mp4, .m4a, and .txt.